client/post-editor/drafts-button/index.jsx

```diff
@@ -1,31 +1,43 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { PropTypes } from 'react';
 
 /**
  * Internal dependencies
  */
-var config = require( 'config' ),
-	Count = require( 'components/count' );
+import Count from 'components/count';
 
-var DraftsButton = React.createClass( {
+export default React.createClass( {
+	displayName: 'EditorDraftsButton',
 
-	render: function() {
-		if ( ! config.isEnabled( 'editor-drafts' ) || ! this.props.site ) {
+	propTypes: {
+		site: PropTypes.object,
+		count: PropTypes.number,
+		onClick: PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			count: 0,
+			onClick: () => {}
+		};
+	},
+
+	render() {
+		if ( ! this.props.site ) {
 			return null;
 		}
 
 		return (
-			<button className="drafts-button" onClick={ this.props.onClick } aria-label={ this.translate( 'View all drafts' ) }>
+			<button className="drafts-button"
+				onClick={ this.props.onClick }
+				disabled={ ! this.props.count }
+				aria-label={ this.translate( 'View all drafts' ) }
+			>
 				<span>{ this.translate( 'Drafts' ) }</span>
-				{ this.props.count ?
-					<Count count={ this.props.count } />
-				: null }
+				{ this.props.count && this.props.count > 0 && <Count count={ this.props.count } /> }
```

Comment:
Still seeing:
<img width="316" alt="drafts0" src="https://cloud.githubusercontent.com/assets/1270189/11644169/ba8325f6-9cfe-11e5-9a8b-a8696f988dc2.png">

To repro: add a single draft. Navigate to `posts/` and delete all drafts. Navigate back to `/post`. 

In this case `0 && false`  will return 0. 

Try using the ternary operator instead: 
```javascript
{ this.props.count && this.props.count > 0 ? <Count count={ this.props.count } /> : null }
```
I tested with the latter and confirmed that it works.